src/widget/wtrackproperty.cpp

```diff
@@ -71,3 +112,203 @@ void WTrackProperty::dragEnterEvent(QDragEnterEvent *event) {
 void WTrackProperty::dropEvent(QDropEvent *event) {
     DragAndDropHelper::handleTrackDropEvent(event, *this, m_pGroup, m_pConfig);
 }
+
+void WTrackProperty::slotOpenInFileBrowser() {
+    QString trackLocation = m_pCurrentTrack->getLocation();
+    mixxx::DesktopHelper::openInFileBrowser(QStringList(trackLocation));
+}
+
+void WTrackProperty::slotPopulatePlaylistMenu() {
```

Comment:
Do not duplicate code from WTrackTableView. The context menu should be factored out into its own class so code is shared between this and WTrackTableView.

Reply:
I don't think we'd want *everything* from WTrackTableView's context menu in WTrackProperty's context menu, for example "Load to", "Hide from library", and maybe the "Add to AutoDJ Queue" actions. So this new class should take a bool argument in its constructor to toggle whether to show those.

Reply:
Thanks. I'll it out into a configurable, pluggable menu.

Reply:
I suggest you use a QFlags/enum class argument instead of boolean instead (like WColorPicker does). It's easier to grasp the meaning of an argument when you read the enum member name instead of `true` in the caller's code. It can also used to make it more fine-grained in case wee need that later on.

Reply:
I was thinking along similar lines. Any widget using the WTrackContextMenu (to be created) can pass a list of enums to include required actions/submenus in the context menu.